api/src/main/java/me/lokka30/treasury/api/economy/currency/CurrencyManager.java

```diff
@@ -65,6 +65,102 @@ public void registerCurrency(@NotNull Currency currency) {
+        Objects.requireNonNull(currencies, "currencies");
+        for (Currency currency : currencies) {
+            registerCurrency(currency);
+        }
+    }
+
+    /**
+     * @param currencies currencies to register
+     * @see #registerCurrency(Currency)
+     */
+    public void registerCurrency(@NotNull Iterable<@NotNull Currency> currencies) {
+        Objects.requireNonNull(currencies, "currencies");
+        for (Currency currency : currencies) {
+            registerCurrency(currency);
+        }
+    }
+
+    /**
+     * Unregisters the specified {@link Currency} {@code currency} if it has been registered.
+     *
+     * @param currency the currency you want unregistered
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public void unregisterCurrency(@NotNull Currency currency) {
+        Objects.requireNonNull(currency, "currency");
+        BY_ID.remove(currency.getCurrencyId());
+        for (String acceptableName : currency.getCurrencyNames()) {
+            BY_NAME.remove(acceptableName);
+        }
+        if (currency.getCurrencyCharacter() != null) {
+            BY_CHAR.remove(currency.getCurrencyCharacter());
+        }
+    }
+
+    /**
+     * @param currencies the currencies you want unregistered
+     * @see #unregisterCurrency(Currency)
+     */
+    public void unregisterCurrency(@NotNull Currency @NotNull ... currencies) {
+        Objects.requireNonNull(currencies, "currencies");
+        for (Currency currency : currencies) {
+            unregisterCurrency(currency);
+        }
+    }
+
+    /**
+     * @param currencies the currencies you want unregistered
+     * @see #unregisterCurrency(Currency)
+     */
+    public void unregisterCurrencies(@NotNull Iterable<@NotNull Currency> currencies) {
+        Objects.requireNonNull(currencies, "currencies");
+        for (Currency currency : currencies) {
+            unregisterCurrency(currency);
+        }
+    }
+
+    /**
+     * Unregisters the {@link Currency} which has been registered with the specified {@link UUID} {@code currencyId} if
+     * it is present.
+     *
+     * @param currencyId the id of the currency you want unregistered
+     * @author MrIvanPlays
+     * @since v1.0.0
+     */
+    public void unregisterCurrency(@NotNull UUID currencyId) {
+        getCurrency(currencyId).ifPresent(this::unregisterCurrency);
+    }
+
+    /**
+     * @param currencyIds currency ids to unregister
+     * @see #unregisterCurrency(UUID)
+     */
+    public void unregisterCurrency(@NotNull UUID @NotNull ... currencyIds) {
+        Objects.requireNonNull(currencyIds, "currencyIds");
+        for (UUID currencyId : currencyIds) {
+            unregisterCurrency(currencyId);
+        }
+    }
+
+    /**
+     * @param currencyIds currency ids to unregister
+     * @see #unregisterCurrency(UUID)
+     */
+    public void unregisterCurrency(@NotNull Iterable<@NotNull UUID> currencyIds) {
```

Comment:
Should this be `unregisterCurrencies`?

Reply:
Because it allows to unregister several currencies at ones